compiler/msgs.nim

```diff
@@ -683,16 +683,43 @@ proc outWriteln*(s: string) =
   if eStdOut in errorOutputs: writeln(stdout, s)
 
 proc msgWriteln*(s: string) =
-  ## Writes to stdout. If --stdout option is given, writes to stderr instead.
+  ## Writes to stdout. If --stderr option is given, writes to stderr instead.
 
   #if gCmd == cmdIdeTools and optCDebug notin gGlobalOptions: return
 
+  var
+    colorStr = s.split(':')[0]
```

Comment:
Meh I dislike parsing the string which can create all sorts of problems. Instead you need to keep the information somehow.